base/ltcmd.dtx

```diff
@@ -620,31 +620,51 @@
 % \end{macro}
 % \end{macro}
 %
-% \begin{macro}{\@@_declare_env:nnnn}
+% \begin{macro}{\@@_declare_env:nnnn, \@@_declare_env:ennn}
+% \changes{v1.2f}{2024/07/06}{Use space-trimmed envname directly}
 % \begin{macro}{\@@_declare_env_internal:nnnn}
+% \begin{macro}{\@@_set_environment_end:n}
 %   The lead-off to creating an environment is much the same as that for
 %   creating a command: issue the appropriate message, store the argument
 %   specification then hand off to an internal function.
 %    \begin{macrocode}
+%<latexrelease>\IncludeInRelease{2024/11/01}{\@@_declare_env:nnnn}%
+%<latexrelease>                 {Use~space-trimmed~envname~directly}
 \cs_new_protected:Npn \@@_declare_env:nnnn #1#2
   {
-    \str_set:Nx \l_@@_environment_str {#1}
-    \str_set:Nx \l_@@_environment_str
-      { \tl_trim_spaces:o { \l_@@_environment_str } }
-    \cs_if_exist:cTF { \l_@@_environment_str }
-      {
-        \msg_info:nnxx { cmd } { redefine-env }
-          { \l_@@_environment_str } { \tl_to_str:n {#2} }
-      }
-      {
-        \msg_info:nnxx { cmd } { define-env }
-          { \l_@@_environment_str } { \tl_to_str:n {#2} }
-      }
+    \str_set:Nn \l_@@_environment_str {#1}
+    \cs_if_exist:cTF { #1 }
+      { \msg_info:nnnn { cmd } { redefine-env } { #1 } { #2 } }
+      { \msg_info:nnnn { cmd } { define-env } { #1 } { #2 } }
     \bool_set_false:N \l_@@_expandable_bool
     \bool_set_true:N \l_@@_environment_bool
-    \exp_args:NV \@@_declare_env_internal:nnnn
-      \l_@@_environment_str {#2}
+    \@@_declare_env_internal:nnnn {#1} {#2}
   }
```

Comment:
A tiny difference: before str (the content of `\l_@@_environment_str`) is used, now tl (the expansion result of `\tl_trim_spaces:e { <envname> }`) is used.

As the space-trimmed `<envname>` is either used in csname or message, this change shouldn't affect behavior of code.